### ISSUE TITLE
[FEATURE] Copy/Paste Table As/From CSV

### ISSUE THREAD

**Issue by nil0x42** [TriliumNext/trilium#1693]
Hi !

## Intro
Ability to do more with tables is a highly requested feature (https://github.com/zadam/trilium/issues/410, https://github.com/zadam/trilium/issues/1586, https://github.com/zadam/trilium/issues/1691, etc).
Therefore most feature request are either very specific, or very hard to implement.

## Feature request
Implement `CSV import from clipboard` and `paste CSV as table` buttons.
This allows user to easily do what they want, handling many, many cases:
* sort/filter/orderBy on a table
  * i can just copy table as CSV, open it with an external software allowing such table manipulations
  * if i want to update result, i can export as CSV from my tool, and paste back to trilium.
  * of course power users are free to automate this by binding shortcuts in their OS to "open table tool with clipboard content as CSV", etc..
* import a table from any software into trilium:
  * `export as CSV` is a feature that almost every table software has, so it should make it trivial
* export a trilium table somewhere else:
  * same as above, `import CSV` is a very common feature in table handling softwares.

## Implementation suggestion
As trilium specific buttons were added to ckeditor's `Edit Block` floating menu, i imagine it would be similar to add custom buttons into ckeditor's `table options menu`:
![Screenshot_2021-02-26_20-48-33](https://user-images.githubusercontent.com/3504393/109353261-0b890280-7874-11eb-98db-56aebd98397c.png)

**Comment by zadam** [comment]
Hi, you can just copy paste tables from Excel or similar software, no need to go through CSV export/import step.

For the other suggestions regarding more features for table I'm skeptical. Table plugin in CKEditor is pretty complex on its own, extending it will not be easy. There's also almost infinite amount of possible things to add there ...

Instead it might be better to implement some ready made advanced table component like handsontable, jExcel or similar. But I'm not yet clear on how exactly to do that (and if at all).

**Comment by SHx7ejs7fCrP** [comment]
i cannot copy tables how do we do that???

**Comment by zadam** [comment]
Yes, Libreoffice Calc should work. Text type is also correct. See this:

![](https://user-images.githubusercontent.com/617641/109398566-f5477900-793d-11eb-94f8-7394c3ca36bb.gif)

**Comment by SHx7ejs7fCrP** [comment]
yea dunno why it wasn't working yesterday, it does work now. 
The thing that separates your wiki night and day is table support, and such great table support...
Wish you would add features to it...

**Comment by plashal** [comment]
+1 on this. csv is such a better format (syntax wise)

**Comment by meichthys** [comment]
[Trilium has entered maintenance mode](https://github.com/zadam/trilium/issues/4620#issuecomment-2094495698). Future enhancements will be addressed in TrilumNext: https://github.com/TriliumNext/trilium/issues/4980

**Comment by eliandoran** [comment]
The new spreadsheet format supports all the requested features here, including CSV import/export.
